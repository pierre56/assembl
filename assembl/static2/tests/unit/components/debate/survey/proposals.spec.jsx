import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Proposals from '../../../../../js/app/components/debate/survey/proposals';

const props = {
  hasPendingPosts: false,
  isPhaseCompleted: false,
  questionIndex: 1,
  questionId: 'question-885432',
  themeId: 'theme-187263',
  title: 'To be or not to be?',
  posts: [
    {
      node: {
        id: 'post-987654',
        originalLocale: 'en'
      }
    }
  ],
  nbPostsToShow: 3,
  phaseUrl: '/my/phase/url',
  questionsLength: 5
};

const renderer = new ShallowRenderer();

describe('Proposals component', () => {
  it('should display a list of proposals without posts', () => {
    const jsx = <Proposals {...props} posts={[]} />;
    renderer.render(jsx);
    const rendered = renderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  it('should display a list of proposals without pending posts', () => {
    const jsx = <Proposals {...props} />;
    renderer.render(jsx);
    const rendered = renderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  it('should display a list of proposals with pending posts', () => {
    const jsx = <Proposals {...props} hasPendingPosts />;
    renderer.render(jsx);
    const rendered = renderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
});