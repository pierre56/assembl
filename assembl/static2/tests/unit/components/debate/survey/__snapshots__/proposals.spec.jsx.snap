// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Proposals component should display a list of proposals with pending posts 1`] = `
<div
  className="shown"
>
  <h3
    className="collapsed-title"
  >
    <span>
      1/ To be or not to be?
    </span>
    <div
      className="shown proposal-arrow"
    >
      <span
        className="assembl-icon-up-open color pointer"
        onClick={[Function]}
      />
    </div>
  </h3>
  <div
    className="shown"
  >
    <Connect(Apollo(Apollo(Apollo(Apollo(Component)))))
      id="post-987654"
      isPhaseCompleted={false}
      originalLocale="en"
      postIndex={0}
      questionId="question-885432"
      themeId="theme-187263"
    />
  </div>
  <div
    className="question-footer"
  >
    <span
      className="all-proposals"
    >
      <Link
        className="button-submit button-light"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/my/phase/url/question/question-885432/1/"
      >
        <Translate
          value="debate.survey.allProposals"
        />
      </Link>
    </span>
  </div>
</div>
`;

exports[`Proposals component should display a list of proposals without pending posts 1`] = `
<div
  className="shown"
>
  <h3
    className="collapsed-title"
  >
    <span>
      1/ To be or not to be?
    </span>
    <div
      className="shown proposal-arrow"
    >
      <span
        className="assembl-icon-up-open color pointer"
        onClick={[Function]}
      />
    </div>
  </h3>
  <div
    className="shown"
  >
    <Connect(Apollo(Apollo(Apollo(Apollo(Component)))))
      id="post-987654"
      isPhaseCompleted={false}
      originalLocale="en"
      postIndex={0}
      questionId="question-885432"
      themeId="theme-187263"
    />
  </div>
  <div
    className="question-footer"
  >
    <span
      className="all-proposals"
    >
      <Link
        className="button-submit button-light"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/my/phase/url/question/question-885432/1/"
      >
        <Translate
          value="debate.survey.allProposals"
        />
      </Link>
    </span>
  </div>
</div>
`;

exports[`Proposals component should display a list of proposals without posts 1`] = `
<div
  className="shown"
>
  <h3
    className="collapsed-title"
  >
    <span>
      1/ To be or not to be?
    </span>
    <div
      className="hidden proposal-arrow"
    >
      <span
        className="assembl-icon-up-open color pointer"
        onClick={[Function]}
      />
    </div>
  </h3>
  <div
    className="question-footer"
  >
    <span
      className="no-proposals"
    >
      <Translate
        value="debate.survey.noProposals"
      />
    </span>
  </div>
</div>
`;
